x-pack/plugins/ml/public/application/components/anomalies_table/links_menu.tsx

```diff
@@ -736,23 +713,46 @@ export const LinksMenuUI = (props: LinksMenuProps) => {
         });
     };
 
-    function findFieldType(index: string) {
-      getFieldTypeFromMapping(index, categorizationFieldName)
-        .then((resp: string) => {
-          if (resp !== '') {
-            createAndOpenUrl(datafeedIndices.join(), resp);
-          } else {
-            i++;
-            if (i < datafeedIndices.length) {
-              findFieldType(datafeedIndices[i]);
-            } else {
-              error();
-            }
-          }
+    // Find the type of the categorization field i.e. text (preferred) or keyword.
+    // Uses the first matching field found in the list of indices in the datafeed_config.
+    let fieldType;
+
+    for (const index of datafeedIndices) {
```

Comment:
This is a regression for the behavior when the datafeed uses a comma-separated style of index pattern. With this change, it no longer finds a mapping for a comma-separated pattern and displays this warning:

<img width="415" alt="Screenshot 2024-05-07 at 16 37 09" src="https://github.com/elastic/kibana/assets/7405507/c9a46fcb-11c3-45ac-9529-e7a4b73f64db">

For this to work, you'll need to see if there is a data view which maps to a datafeed config such as

```
    "indices": [
      "gallery-2023-01",
      "gallery-2023-02",
      "gallery-2023-03"
    ],
```

Reply:
We have a function that might be useful here.
[getDataViewIdFromDatafeed](https://github.com/elastic/kibana/blob/3fe28cb4b2d9578013f1d3908603e5b92f1dde47/x-pack/plugins/ml/public/application/jobs/new_job/pages/index_or_search/preconfigured_job_redirect.ts#L77)

Reply:
James, thanks for the hint about `getDataViewIdFromDataFeed`. I ended up just copying over the approach because that function returns just the id and not the full data view. This allowed me to fix the issue Pete found in [1adabee](https://github.com/elastic/kibana/pull/182588/commits/1adabeedf0c677084bba97d00cfa274ca2948bdb).